content/post/hugo_ga_preview.adoc

```diff
@@ -0,0 +1,218 @@
+----
+    - name: Install Ruby Dev                     
+      run: sudo apt-get install ruby-dev
+
+    - name: Install AsciiDoctor and Rouge
+      run: sudo gem install asciidoctor rouge
+
+    - name: Setup Hugo                           
+      uses: peaceiris/actions-hugo@v2
+      with:
+        hugo-version: '0.85.0'
+----
+
+=== Determining the preview URL
+
+To deploy to `surge.sh` I used the same GitHub Action as Gunnar, which is https://github.com/afc163/surge-preview[this one]. The URL that it deploys to is determined by the source repository and PR number: 
+
+> `https://{{repository.owner}}-{{repository.name}}-{{job.name}}-pr-{{pr.number}}.surge.sh`
+
+These break down as follows: 
+
+* `repository.owner`: `rmoff`
+* `repository.name`: `rmoff-blog`
+* `job.name`: This is the name of the job, specified under `jobs:` in the YAML that defines the Workflow. 
+* `pr.number`: The PR number, which can be obtained from a variable in the workflow. 
+
+I'm stealing https://github.com/gunnarmorling/morling.dev/blob/master/.github/workflows/preview.yml#L17[Gunnar's nice bit of code] here to determine the PR number :) 
+
+[source,yaml]
+----
+    - name: Setup base URL env var
+      run: |
+        export PRNUMBER=$(jq --raw-output .pull_request.number "$GITHUB_EVENT_PATH") 
+        echo BASEURL="https://rmoff-rmoff-blog-preview-pr-"$PRNUMBER".surge.sh/" >> $GITHUB_ENV
+----
+
+Note that `export` works locally to a step, but to pass the variable between steps you append the setting of it to `$GITHUB_ENV`
+
+=== Build the Hugo site
+
+Install of simply `hugo` which builds the site, we add a few command line flags to build draft posts (`--buildDrafts`), future-dated posts (`--buildFuture`), and also change the base URL (`--baseURL`) which by default (per the https://github.com/rmoff/rmoff-blog/blob/master/config.yaml[`config.yaml`]) is the live site (`rmoff.net`). 
+
+Note that if the base URL is incorrectly set the resulting built site won't render properly - you'll get missing CSS files, weird relative links, etc. 
+
+=== Deploying the built site to surge.sh
+
+https://github.com/afc163/surge-preview[This action] deploys the built site (under `/public`, specified in the `dist` parameter) from the previous step to surge.sh: 
+
+[source,yaml]
+----
+    - name: Preview
+      uses: afc163/surge-preview@v1
+      id: preview_step
+      with:
+        surge_token: ${{ secrets.SURGE_TOKEN }}
+        github_token: ${{ secrets.GITHUB_TOKEN }}
+        dist: public
+        failOnError: 'true'
+        teardown: true
+        build: |
+          echo Deploying to surge.sh
+----
+
+== The finished workflow
+
+[source,yaml]
+----
+include::https://raw.githubusercontent.com/rmoff/rmoff-blog/master/.github/workflows/preview-blog.yaml[]
+----
+
+Put this YAML in a file in your github repository root folder under `.github/workflows/`. 
+
+== The workflow in action
+
+1. Create a branch of the master branch, into which you will commit a new blog post
```

Comment:
```suggestion
1. Create a branch of the main branch, into which you will commit a new blog post:
```

Not sure what you have, but I'd show the default name in current versions and on GitHub since recently.

Reply:
yeah it's `master`, but fair point